frontend/src/metabase/dashboard/components/DashboardHeader.tsx

```diff
@@ -85,20 +85,14 @@ const DashboardHeader = ({
   }, [isModalOpened]);
 
   const _headerButtons = useMemo(
-    () =>
-      headerButtons.map((section, sectionIndex) => {
-        return (
-          section.length > 0 && (
-            <HeaderButtonSection
-              key={sectionIndex}
-              className="Header-buttonSection"
-              isNavBarOpen={isNavBarOpen}
-            >
-              {section}
-            </HeaderButtonSection>
-          )
-        );
-      }),
+    () => (
```

Comment:
Previously, it was not obvious what we do here. The `headerButtons` looked like this.

```ts
[[button1, button2, button3]]
```

You could see that we only sent `buttons` in an array making it an array within another array
https://github.com/metabase/metabase/blob/a26cba2a44eae19b92cda37356f28fd9cf6f9af3/frontend/src/metabase/dashboard/containers/DashboardHeader.jsx#L370.

This is simplified now. https://github.com/metabase/metabase/pull/23833/files#diff-353d899c9d83acd25f94eddc7773d8377fbe420ed1bae75084691d0d292f2210R363